File: Wordhunt Solver/solver.py
def create_grid(letters):
    grid = []
    for i in range(0, len(letters), 4):
        row = letters[i:i+4]
        grid.append(list(row))
    return grid

def load_words():
    with open("Wordhunt Solver/words.txt", "r") as f:
        words = [line.strip() for line in f]
    return set(words)

def find_words(grid, words):
    results = []
    for word in words:
        for i in range(4):
            for j in range(4):
                if grid[i][j] == word[0]:
                    if check_word(grid, word, i, j, set()):
                        results.append(word)
    results.sort(key=lambda x: len(x), reverse=True)
    return results

def check_word(grid, word, x, y, visited):
    if not word:
        return True
    if x < 0 or x >= 4 or y < 0 or y >= 4:
        return False
    if (x, y) in visited:
        return False
    if grid[x][y] != word[0]:
        return False
    visited.add((x, y))
    for i in range(-1, 2):
        for j in range(-1, 2):
            if i == 0 and j == 0:
                continue
            if check_word(grid, word[1:], x+i, y+j, visited):
                return True
    visited.remove((x, y))
    return False

print("")

letters = input("Enter your letters here: ")
grid = create_grid(letters)
words = load_words()
results = find_words(grid, words)

print("Results:")
if results:
    results.sort(key=lambda x: len(x), reverse=False)
    for result in results:
        print(result)
else:
    print("No words found.")

print("")
print("↑↑↑↑↑↑ Results ↑↑↑↑↑↑")
print("")
print("-------------------------")
print("")
print("↓↓↓↓↓↓ Grid ↓↓↓↓↓↓")
print("")
for row in grid:
    print(row)
print("")

File: Wordhunt Solver/solver.java
// java implementation because why not
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class WordHuntSolver {

    public static char[][] createGrid(String letters) {
        char[][] grid = new char[4][4];
        for (int i = 0; i < letters.length(); i += 4) {
            String row = letters.substring(i, Math.min(i + 4, letters.length()));
            for (int j = 0; j < row.length(); j++) {
                grid[i / 4][j] = row.charAt(j);
            }
        }
        return grid;
    }

    public static Set<String> loadWords() throws IOException {
        Set<String> words = new HashSet<>();
        try (BufferedReader br = new BufferedReader(new FileReader("Wordhunt Solver/words.txt"))) {
            String line;
            while ((line = br.readLine()) != null) {
                words.add(line.trim());
            }
        }
        return words;
    }

    public static List<String> findWords(char[][] grid, Set<String> words) {
        List<String> results = new ArrayList<>();
        for (String word : words) {
            for (int i = 0; i < 4; i++) {
                for (int j = 0; j < 4; j++) {
                    if (grid[i][j] == word.charAt(0)) {
                        if (checkWord(grid, word, i, j, new HashSet<>())) {
                            results.add(word);
                        }
                    }
                }
            }
        }
        results.sort((a, b) -> Integer.compare(b.length(), a.length()));
        return results;
    }

    public static boolean checkWord(char[][] grid, String word, int x, int y, Set<String> visited) {
        if (word.isEmpty()) {
            return true;
        }
        if (x < 0 || x >= 4 || y < 0 || y >= 4) {
            return false;
        }
        String pos = x + "," + y;
        if (visited.contains(pos)) {
            return false;
        }
        if (grid[x][y] != word.charAt(0)) {
            return false;
        }
        visited.add(pos);
        for (int i = -1; i <= 1; i++) {
            for (int j = -1; j <= 1; j++) {
                if (i == 0 && j == 0) {
                    continue;
                }
                if (checkWord(grid, word.substring(1), x + i, y + j, visited)) {
                    return true;
                }
            }
        }
        visited.remove(pos);
        return false;
    }

    public static void main(String[] args) throws IOException {
        System.out.println("");

        String letters = "Enter your letters here: ";
        char[][] grid = createGrid(letters);
        Set<String> words = loadWords();
        List<String> results = findWords(grid, words);

        System.out.println("Results:");
        if (!results.isEmpty()) {
            results.sort(String::compareTo);
            for (String result : results) {
                System.out.println(result);
            }
        } else {
            System.out.println("No words found.");
        }

        System.out.println("");
        System.out.println("↑↑↑↑↑↑ Results ↑↑↑↑↑↑");
        System.out.println("");
        System.out.println("-------------------------");
        System.out.println("");
        System.out.println("↓↓↓↓↓↓ Grid ↓↓↓↓↓↓");
        System.out.println("");
        for (char[] row : grid) {
            for (char c : row) {
                System.out.print(c + " ");
            }
            System.out.println();
        }
        System.out.println("");
    }
}

